addons/mail/static/src/models/chat_window/chat_window.js

```diff
@@ -150,13 +150,14 @@ registerModel({
                 isMemberListOpened: true,
             });
         },
-        /**
-         * @param {Event} ev
-         */
-        onFocusInNewMessageFormInput(ev) {
-            if (this.exists()) {
-                this.update({ isFocused: true });
+        onComponentUpdate() {
+            if (this.isDoFocus) {
+                this.update({ isDoFocus: false });
+                if (this.newMessageFormInputView) {
+                    this.newMessageFormInputView.doFocus();
+                }
```

Comment:
Shouldn't input view take care of its dofocus itself? It looks like the field is on the wrong model here.

Reply:
Kinda. I think functionally we want to focus the chat window. Focusing the input is a part of focusing the chat window.

There are in-and-out between `ChatWindow`, `ChatWindowComponent`, `AutocompleteInputView` and `AutocompleteInputComponent`.
I think this make sense with current split between components/models and that the macro-goal is focusing chat window (and focusing input being a part of it)

Reply:
But it's wrong in terms of component lifecycle to do it this way. The focusing on children should be done on patch of children, not on patch of parent which can happen at different timing and for different reasons.